electrum/lnworker.py

```diff
@@ -1347,6 +1337,8 @@ def handle_error_code_from_failed_htlc(
         # for errors that do not include a channel update
         else:
             self.network.path_finder.liquidity_hints.add_to_blacklist(fallback_channel)
+        # remove inflight htlcs from liquidity hints
+        self.network.path_finder.update_htlcs_liquidity_hints(route, add_htlcs=False)
```

Comment:
I think this needs to be done earlier in the method as otherwise we might have already raised before getting here.